package_osx.sh copy deps from homebrew and mods from flare-game

ludoza opened the pull request:
With this changes `package_osx.sh` can now be called as normal to only package the engine and collect dependencies from website:

```sh
$ ./package_osx.sh flare
download dependencies from website
...
--2019-09-05 11:56:20--  http://files.ruads.org/flare_osx_dependencies.tar.gz
Resolving files.ruads.org (files.ruads.org)... 5.9.52.199
Connecting to files.ruads.org (files.ruads.org)|5.9.52.199|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4969402 (4.7M) [application/octet-stream]
...
2019-09-05 11:56:22 (2.91 MB/s) - ‘/tmp/___flare.build/flare_osx_dependencies.tar.gz’ saved [4969402/4969402]

packaging
done
```

Or package the engine with dependencies from homebrew:

```sh
$ ./package_osx.sh flare homebrew
copy dependencies from homebrew
packaging
done
```

And finally to package the engine with mods specified in `flare-engine/mods/mods.txt`:

```sh
 ./package_osx.sh flare homebrew ../flare-game/ # or `./package_osx.sh flare http ../flare-game/`
copy dependencies from homebrew
copied fantasycore
copied empyrean_campaign
packaging
done
```

I'm not sure why the website dependancies have the `*.a` files and `libfreetype`, because it does not seem to be needed and just add bloat to the `lib/` directory, but I can just be naive, so will ask in issue #1583.

After packaging using homebrew I did uninstall all dependancies and tested the package: 

```sh
brew uninstall --ignore-dependencies cmake libvorbis sdl2 sdl2_image sdl2_mixer sdl2_ttf libogg
```

I uploaded the packages here(sorry for using a free service, hope it works):

- ~~[flare_osx-engine-190905.tar.gz]() (5.5M MD5 10de931da8d4d0d51f36a21ac137b2ed)~~
- ~~[flare_osx-game-190905.tar.gz]() (124M MD5 7db07d162d563d6eb9f1465621e3cc62)~~

Let me know if I need to change something.

dorkster commented:
I was going to say `libfreetype` was required for SDL_ttf. However, it looks like the SDL_ttf devs bundle FreeType with their lib. Thus explaining why it's not necessary to get it from an external source. I assume the builds you've made without `libfreetype` run.

From what I understand, the `*.a` files are the static versions of the libraries, whereas the `*.dylib` ones are dynamic. So I don't see the need for including the `*.a` files, but I'd like some confirmation before merging.

Other than that, this script looks good.

ludoza commented:
>  I assume the builds you've made without libfreetype run.

Yes, I tested it on my local machine after uninstalling the used homebrew packages. But would also like some confirmation from another macos user.

Thanks for the explanation on the [`ar`](https://en.wikipedia.org/wiki/Ar_(Unix)) files. I was more questioning the need for them in the distributed package. Because if they were used they would be statically linked into the `flare` binary. And in that case they will also not need to be distributed?

dorkster commented:
>Because if they were used they would be statically linked into the flare binary. And in that case they will also not need to be distributed?

That sounds correct to me.

>Yes, I tested it on my local machine after uninstalling the used homebrew packages. But would also like some confirmation from another macos user.

Great. I agree that some confirmation from another user would be nice before merging this.

ludoza commented:
Ever found a new hammer and all your problems suddenly looks like nails? Well that is me with gitlab-runner and continuous integration.

I created a [.gitlab-ci.yml](https://gitlab.com/profhound/flare-macos/blob/master/.gitlab-ci.yml) script that [compiles flare](https://gitlab.com/profhound/flare-macos/-/jobs/289614328) (only the engine atm) and then also [create a dmg](https://gitlab.com/profhound/flare-macos/-/jobs/289614331).

The nice part is it uploads the results as artifacts to gitlab that other people can download from instead of the free file hosting site I stumbled upon.  

- [flare_engine.dmg](https://gitlab.com/profhound/flare-macos/-/jobs/289614331/artifacts/browse)

The `Flare.app` icon is still broken so I'm going to try to fix that. And then also create a `flare_game.dmg`.

If that is successful I will solicit some social networks and see if I can find us a macos tester or two.

To confirm the build works on their macos.

Later,

ludoza commented:
_Wow_, to get the icons correct is a lot more painful than expected. No wonder the Mac OS X maintainer thread is close to a 100 comments!1

`create-dmg` seems to use a AppleScript to start Finder to generate layout and icon information. But only sleep 1 sec to generate the meta data. So that is why it works on a small dmg `flare_engine` but not on `flare_game`.

I then tried `dmgbuild` which I had to fork and patch to work on python3. And while I could generate correct icons by using my script by hand, the continuous integration does not seem to be able to do it.

I'm kind of going to give up on it for now and just see if the [`flare_game.dmg`](https://gitlab.com/profhound/flare-macos/-/jobs/289876875/artifacts/raw/flare_game.dmg) (123 MB) works for other people on their mac with the minimum set of `dylib`s.

I will report back to this pull request once I have more feedback.

elosha commented:
Mind using this one? ;)
[flare.icns.zip](https://github.com/flareteam/flare-engine/files/3586564/flare.icns.zip)

ludoza commented:
@elosha thanks for the icon, I added it to the flare-dmg project.

As mentioned I fixed missing dependencies and if I build the `flare.dmg` by hand the icon display correctly:

- [flare.dmg](https://gofile.io/?c=cNbrDt) (124MB MD5 03ab312f56033899c768e722b56fe83a)

I'm waiting for the CI process to finish in hopes your icon resolved my issue. But I seem not to be the only one on the internet having troubles with building a dmg in a headless mode.

ludoza commented:
Just some feedback got 3 people to test the above dmg and they all said it works:

- 10.14.6
- High Sierra version 10.13.6
- Catalina
